core/src/databases_store/gcs.rs

```diff
@@ -154,12 +166,20 @@ impl DatabasesStore for GoogleCloudStorageDatabasesStore {
                 .collect::<HashMap<_, _>>();
 
             // Merge the two maps to get the final rows.
-            // New ones take precedence.
+            // New ones take precedence, which includes the case where a 'delete' row
+            // replaces a regular row.
```

Comment:
This needs to be in a spawn_blocking I think

Reply:
Done.